receiver/rabbitmqreceiver/scraper.go

```diff
@@ -70,18 +70,48 @@ func (r *rabbitmqScraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
 		return pmetric.NewMetrics(), errClientNotInit
 	}
 
-	// Get queues for processing
+	var errs []error // Collect multiple errors instead of failing immediately
+
+	// Collect queue metrics
+	if err := r.collectQueueMetrics(ctx, now); err != nil {
+		errs = append(errs, err)
+	}
+
+	// Collect node metrics
+	if err := r.collectNodeMetrics(ctx, now); err != nil {
+		errs = append(errs, err)
+	}
+
+	// If there are errors, return them as a combined error
+	if len(errs) > 0 {
+		return pmetric.NewMetrics(), errors.Join(errs...)
```

Comment:
you still need to emit the metrics or you will continue to accumulate them iirc.

This is what we do for oracle:
```
if len(scrapeErrors) > 0 {
		return out, scrapererror.NewPartialScrapeError(multierr.Combine(scrapeErrors...), len(scrapeErrors))
	}
	return out, nil
```